toolchain/sem_ir/function.h

```diff
@@ -65,10 +65,18 @@ struct Function : public EntityWithParamsBase,
   }
 
   // Given a parameter reference instruction from `param_refs_id` or
-  // `implicit_param_refs_id`, returns the corresponding `Param` instruction
-  // and its ID.
+  // `implicit_param_refs_id`, returns a `ParamInfo` value with the
+  // corresponding instruction, its ID, and the name binding, if present.
+  struct ParamInfo {
+    InstId inst_id;
+    std::optional<AnyBindName> bind_name;
+    Param inst;
```

Comment:
It seems cleaner and less surprising to have `inst_id` right next to `inst`.

```suggestion
    Param inst;
    std::optional<AnyBindName> bind_name;
```